components/suggest/src/db.rs

```diff
@@ -216,12 +223,42 @@ impl<'a> SuggestDao<'a> {
                                         number_of_ratings: row.get("number_of_ratings")?,
                                         guid: row.get("guid")?,
                                         score: row.get("score")?,
-                                    })
+                                    }))
                             })
                     },
+                    SuggestionProvider::Pocket => {
+                        let confidence = row.get("confidence")?;
+                        let suffixes_match = match confidence {
+                            KeywordConfidence::Low => row.get::<_, String>("keyword_suffix")?.starts_with(keyword_suffix),
+                            KeywordConfidence::High => row.get::<_, String>("keyword_suffix")? == keyword_suffix,
+                        };
+                        if suffixes_match {
+                            self.conn.query_row_and_then(
+                                "SELECT p.score
+                                FROM pocket_custom_details p
+                                WHERE p.suggestion_id = :suggestion_id",
+                                named_params! {
+                                    ":suggestion_id": suggestion_id
+                                },
+                                |row| {
+                                    Ok(Some(Suggestion::Pocket {
+                                        title,
+                                        url: raw_url,
+                                        score: row.get("score")?,
+                                        is_top_pick: matches!(
+                                            confidence,
+                                            KeywordConfidence::High
+                                        )
+                                    }))
+                                }
+                            )
+                        } else {
+                            Ok(None)
+                        }
+                    }
                 }
-            },
-        )
+            }
+        )?.into_iter().flatten().collect())
```

Comment:
Discussion: we could provide some extra convenience by sorting the returning suggestions by certain criteria, e.g. by score or rank. For example, if a keyword matches both the high confidence and the low confidence keywords for a Pocket suggestion, it'll be nice to put the high confidence one in the front.